.github/workflows/test-coverage.yml

```diff
@@ -0,0 +1,430 @@
+                    --buildtype      ${{ matrix.config }}
+                    --platform       ${{ matrix.image }}
+                    --incremental    ${{ inputs.incremental }}
+                    --base_ref       ${{ inputs.base_ref }}
+                    --head_ref       ${{ inputs.head_ref }}
+                    --repository     ${{ github.server_url }}/${{ github.repository }}
+              "
+
+      - name: Nightly build
+        if:   github.event_name == 'schedule'
+        run: ".github/workflows/root-ci-config/build_root.py
+                    --buildtype      ${{ matrix.config }}
+                    --platform       ${{ matrix.image }}
+                    --incremental    false
+                    --base_ref       ${{ github.ref_name }}
+                    --repository     ${{ github.server_url }}/${{ github.repository }}
+              "
+            
+
+
+      - name: Update artifacts after push to release branch
+        if:   github.event_name == 'push'
+        run: ".github/workflows/root-ci-config/build_root.py
+                    --buildtype      ${{ matrix.config }}
+                    --platform       ${{ matrix.image }}
+                    --incremental    false
+                    --base_ref       ${{ github.ref_name }}
+                    --repository     ${{ github.server_url }}/${{ github.repository }}
+              "
+    
+      - name: Upload test results
+        if:   ${{ !cancelled() }}
+        uses: actions/upload-artifact@v3
+        with:
+          name:  Test Results ${{ matrix.image }} ${{ matrix.config }}
+          path: /github/home/ROOT-CI/build/TestResults.xml
+      
+      # - name: Upload coverage results
+      #   if:   ${{ matrix.image == 'fedora38-cov' && !cancelled() }}
+      #   uses: actions/upload-artifact@v3
+      #   with:
+      #     name: Test Coverage ${{ matrix.image }} ${{ matrix.config }}
+      #     path: /tmp/workspace/build/CoverageXML.xml
+      # - name: Setup tmate session
+      #   if: failure() && runner.debug == '1'
+      #   uses: mxschmitt/action-tmate@v3
+      #   with:
+      #     sudo: false
+      #   # When debugging increase to a suitable value!
+      #   timeout-minutes: ${{ github.event.pull_request && 40 || 20 }}
+
+      - name: Upload coverage to Codecov
+        if: ${{ matrix.image == 'fedora38-cov' && !cancelled() }}
+        uses: codecov/codecov-action@v3
+        with:
+          env_vars: OS,PYTHON
+          fail_ci_if_error: true
+          files: /github/home/ROOT-CI/build/CoverageXML.xml
+          flags: unittests
+          name: codecov-umbrella
+          verbose: true   
+  
+  event_file:
+    if: github.repository_owner == 'root-project' || github.event_name == 'pull_request'
+    name: "Upload Event Payload"
+    runs-on: ubuntu-latest
+    steps:
+    - name: Upload
+      uses: actions/upload-artifact@v3
+      with:
+        name: Event File
+        path: ${{ github.event_path }}
```

Comment:
I don't think we need this section. But we do want a newline at the end of the file :-)